db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1627 @@
+        statement.get().executeUpdate(
+            // language=sql
+            """
+            INSERT INTO
+              merlin.external_source
+            VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s')
+            """.formatted(
+            failing.key(),
+            failing.source_type_name(),
+            failing.derivation_group_name(),
+            failing.valid_at(),
+            failing.start_time(),
+            failing.end_time(),
+            failing.created_at(),
+            failing.metadata()
+          )
+        );
+      });
+
+      // upload a non-conflicting source (same name in a different dg)
+      assertDoesNotThrow(() -> {
+        statement.get().executeUpdate(
+            // language=sql
+            """
+            INSERT INTO merlin.derivation_group VALUES ('%s', '%s')
+            """.formatted(dg + "_2", st)
+        );
+        statement.get().executeUpdate(
+            // language=sql
+            """
+            INSERT INTO
+              merlin.external_source
+            VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s')
+            """.formatted(
+            succeeding.key(),
+            succeeding.source_type_name(),
+            succeeding.derivation_group_name(),
+            succeeding.valid_at(),
+            succeeding.start_time(),
+            succeeding.end_time(),
+            succeeding.created_at(),
+            succeeding.metadata()
+          )
+        );
+      });
+    }
+
+    /**
+     * No two derivation groups, even across different source types, can have the same name.
+     */
+    @Test
+    void duplicatedDG() {
+      final var statement = getStatement();
+
+      assertDoesNotThrow(() -> {
+        // create a derivation group of the Test type
+        merlinHelper.upload_source(dg);
+        statement.set(connection.createStatement());
+        statement.get().executeUpdate("INSERT INTO merlin.external_source_type VALUES ('New Name');");
+      });
+
+      assertThrowsExactly(PSQLException.class, () -> {
+        // use the same name as before (Test Default) with a different source type (New Name) - fails
+        // (This is noteworthy as this is newer behavior.)
+        statement.get().executeUpdate("INSERT INTO merlin.derivation_group VALUES ('%s', 'New Name');".formatted(dg));
+      });
+    }
+
+    /**
+     * It is impossible to delete a derivation group if there is an entry in the plan-derivation group link table with
+     *    it. This is caught explicitly in the frontend (aerie-ui), but for alternative frontends like a CLI, we should
+     *    catch it at this (PSQL) level. We demonstrate that here.
+     */
+    @Test
+    void deleteDGwithRemainingPlanLink() {
+      final var statement = getStatement();
+
+      // create all
+      assertDoesNotThrow(() -> {
+        statement.set(connection.createStatement());
+
+        // create plan with minimal input (doesn't require a model!)
```

Comment:
Please don't do this. Instead, use the proper style of inserting a model and plan using the `MerlinDBTestsHelper` methods.